src/modules/FunModule.cpp

```diff
@@ -615,19 +667,67 @@ FunModule::FunModule(UmikoBot* client) : Module("funutil", true), m_memeChannel(
 
 		m_GithubChannel = channel.id();
 
-		if(args.size() != 1)
+		if(args.size() > 3)
 		{
 			UmikoBot::Instance().createMessage(m_GithubChannel, "**Wrong Usage of Command!**");
 
 			return;
 		}
 
-		// For extra randomness
-		std::random_device device;
-		std::mt19937 rng(device());
-		std::uniform_int_distribution<std::mt19937::result_type> ch('A', 'Z');
+		QString repo;
+
+		m_GithubFlag = GITHUB_RANDOM;
+
+		if(args.size() == 2)
+		{
+			m_GithubFlag = GITHUB_REPOSITORY;
+		}
+
+		for(int i = 1; i < args.size(); i++)
+		{
+			auto text = args[i].toLower();
+
+			if(text == "--random")
+			{
+					m_GithubFlag = GITHUB_RANDOM;
+
+					continue;
+			}
+			else if(text == "--repository")
+			{
+					m_GithubFlag = GITHUB_REPOSITORY;
+
+					continue;
+			}
+			else if(text == "--search")
+			{
+					m_GithubFlag = GITHUB_SEARCH;
 
-		m_GithubManager.get(QNetworkRequest(QUrl("https://api.github.com/search/repositories?q=" + QString(QChar((char)ch(rng))))));
+					continue;
+			}
+			else
+			{
+				repo = text;
+			}
+		}
+
+		if(m_GithubFlag == GITHUB_RANDOM)
+		{
+			// For extra randomness
+			std::random_device device;
+			std::mt19937 rng(device());
+			std::uniform_int_distribution<std::mt19937::result_type> ch('A', 'Z');
+
+			m_GithubManager.get(QNetworkRequest(QUrl("https://api.github.com/search/repositories?q=" + QString(QChar((char)ch(rng))))));
+		}
+		else if(m_GithubFlag == GITHUB_SEARCH)
+		{
+			m_GithubManager.get(QNetworkRequest(QUrl("https://api.github.com/search/repositories?q=" + repo)));
+		}
+		else if(m_GithubFlag == GITHUB_REPOSITORY)
+		{
+			m_GithubManager.get(QNetworkRequest(QUrl("https://api.github.com/repos/" + repo)));
```

Comment:
This `get` function returns a `QNetworkReply*` that you can use to listen to its `finished` signal, instead of shoving all functionality in the QNetworkAccessManager's finished signal!